include/drivers/ipm.h

```diff
@@ -232,6 +367,58 @@ static inline int z_impl_ipm_set_enabled(const struct device *ipmdev,
 	return api->set_enabled(ipmdev, enable);
 }
 
+/**
+ * @brief Return the maximum number of channels.
+ *
+ * Return the maximum number of channels supported by the hardware.
+ *
+ * @param ipmdev Driver instance pointer.
+ *
+ * @return Maximum possible number of supported channels.
+ */
+__syscall uint32_t ipm_max_channels_get(const struct device *ipmdev);
+
+static inline uint32_t z_impl_ipm_max_channels_get(const struct device *ipmdev)
+{
+	const struct ipm_driver_api *api =
+		(const struct ipm_driver_api *)ipmdev->api;
+
+	if (api->max_channels_get == NULL) {
+		/* At least 1 channel */
+		return 1;
+	}
+
+	return api->max_channels_get(ipmdev);
+}
+
+/**
+ * @brief Configure inbound channels.
+ *
+ * Enable interrupts or setup the device for the specified inbound channels.
+ *
+ * @param ipmdev Driver instance pointer.
+ * @param channel IPM device channel
+ * @param enable Set to 0 to disable and to nonzero to enable.
+ *
+ * @retval 0       On success.
+ * @retval -EINVAL If it isn't an inbound channel.
+ */
+__syscall int ipm_configure_channel(const struct device *ipmdev, uint32_t channel,
+				    int enable);
```

Comment:
Wonder if it would be better to pass a `uint32_t` instead of a simple `enable`. We can reserve, for example, the first 8-bit for common IPM configurations, and the rest would be up to individual drivers. For now, the common part would only have 1-bit of enabling and disabling the channel, and some bits are reserved for future expansions.

Reply:
That looks like a good idea and if we have to touch the API better doing all at once. I'll prepare the change.

Reply:
I changed `s/int enable/uint32_t conf`